search/indexing_api.py

```diff
@@ -397,167 +410,107 @@ def update_post(doc_id, post):
-            doc_type=GLOBAL_DOC_TYPE,
-            # Adjust chunk size from 500 depending on environment variable
-            chunk_size=settings.ELASTICSEARCH_INDEXING_CHUNK_SIZE,
-        )
-        if len(errors) > 0:
-            raise ReindexException(
-                "Error during bulk profile insert: {errors}".format(errors=errors)
-            )
+    index_items(serialize_bulk_profiles, PROFILE_TYPE, ids)
 
 
 def index_courses(ids):
     """
-    Index courses based on list of course ids
+    Index a list of courses by id
 
     Args:
-        ids(list of int): List of course id's
+        ids(list of int): List of Course id's
     """
-    conn = get_conn()
-    for alias in get_active_aliases([COURSE_TYPE]):
-        _, errors = bulk(
-            conn,
-            serialize_bulk_courses(ids),
-            index=alias,
-            doc_type=GLOBAL_DOC_TYPE,
-            # Adjust chunk size from 500 depending on environment variable
-            chunk_size=settings.ELASTICSEARCH_INDEXING_CHUNK_SIZE,
-        )
-        if len(errors) > 0:
-            raise ReindexException(
-                "Error during bulk course insert: {errors}".format(errors=errors)
-            )
+    index_items(serialize_bulk_courses, COURSE_TYPE, ids)
 
 
 def index_bootcamps(ids):
     """
-    Index bootcamps based on list of bootcamp ids
+    Index a list of bootcamps by id
 
     Args:
-        ids(list of int): List of bootcamp id's
+        ids(list of int): List of Course id's
     """
-    conn = get_conn()
-    for alias in get_active_aliases([BOOTCAMP_TYPE]):
-        _, errors = bulk(
-            conn,
-            serialize_bulk_bootcamps(ids),
-            index=alias,
-            doc_type=GLOBAL_DOC_TYPE,
-            # Adjust chunk size from 500 depending on environment variable
-            chunk_size=settings.ELASTICSEARCH_INDEXING_CHUNK_SIZE,
-        )
-        if len(errors) > 0:
-            raise ReindexException(
-                "Error during bulk bootcamp insert: {errors}".format(errors=errors)
-            )
+    index_items(serialize_bulk_bootcamps, BOOTCAMP_TYPE, ids)
 
 
 def index_programs(ids):
     """
-    Index programs based on list of program ids
+    Index a list of programs by id
 
     Args:
-        ids(list of int): List of program id's
+        ids(list of int): List of Course id's
     """
-    conn = get_conn()
-    for alias in get_active_aliases([PROGRAM_TYPE]):
-        _, errors = bulk(
-            conn,
-            serialize_bulk_programs(ids),
-            index=alias,
-            doc_type=GLOBAL_DOC_TYPE,
-            # Adjust chunk size from 500 depending on environment variable
-            chunk_size=settings.ELASTICSEARCH_INDEXING_CHUNK_SIZE,
-        )
-        if len(errors) > 0:
-            raise ReindexException(
-                "Error during bulk program insert: {errors}".format(errors=errors)
-            )
+    index_items(serialize_bulk_programs, BOOTCAMP_TYPE, ids)
```

Comment:
`PROGRAM_TYPE`

Reply:
And needs test

Reply:
weird, I recreated the index but still have 9 program results

Reply:
Oh, it gets saved in the wrong index, but still shows up in the all index alias used for search